src/lib/containers/widgets/query-filter/QueryFilter.tsx

```diff
@@ -138,22 +139,40 @@ export const applyChangesToRangeColumn = (
 export const QueryFilter: React.FunctionComponent<QueryFilterProps> = ({
   facetsToFilter,
 }): JSX.Element => {
+  const [facets, setFacets] = React.useState<FacetColumnResult[]>([])
+  const [facetFilter, setFacetFilter] = React.useState(facetsToFilter ?? [])
+  const { facetAliases, topLevelControlsState } = useQueryVisualizationContext()
+  const { showFacetFilter } = topLevelControlsState
   const { data, isLoadingNewBundle, getLastQueryRequest, executeQueryRequest } =
     useQueryContext()
 
-  const { facetAliases, topLevelControlsState } = useQueryVisualizationContext()
+  React.useEffect(() => {
+    if (data) {
+      setFacets(data.facets as FacetColumnResult[])
+    }
+  }, [])
+
+  React.useEffect(() => {
+    if (data) {
+      const facets = data.facets as FacetColumnResult[]
+      if (facetFilter.length > 0) {
+        let newArr = facets.filter(facet => {
+          return facetsToFilter
+            ? facetFilter.includes(facet.columnName)
+            : !facetFilter.includes(facet.columnName)
+        })
+        setFacets(newArr)
+      } else {
+        setFacets(facets)
+      }
+    }
+  }, [facetFilter])
```

Comment:
This doesn't seem right, but I want to wait for the other changes before digging into it